src/python/WMCore/WMRuntime/NATSInterface.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+"""
+NATSInfo provides wrapper around NATS to send WMAgent messages
+"""
+from __future__ import division
+from __future__ import print_function
+
+# system modules
+import socket
+
+# WMCore modules
+from WMCore.WMSpec.WMStep import WMStepHelper
+from WMCore.WMSpec.WMWorkload import getWorkloadFromTask, WMWorkloadHelper
+from WMCore.Storage.SiteLocalConfig import loadSiteLocalConfig, SiteConfigError
+from WMCore.WMRuntime.Bootstrap import getSyncCE
+
+# CMSMonitoring modules
+from CMSMonitoring.NATS import NATSManager
+
+
+class NATSInfo(object):
```

Comment:
Should we rename it to something like `NATSPublisher`? The file is named as `NATSInterface`, which clearly it's not! NATSInfo doesn't seem to be the best choice either.

Reply:
For this round, I changed NATSInfo to NATSPublisher, but I kept NATSInterface as a module name, since Publisher may be just one of the interfaces. If you still be in favor of new name we can rename it.